src/vos/vos_io.c

```diff
@@ -1905,16 +1889,35 @@ reserve_space(struct vos_io_context *ioc, uint16_t media, daos_size_t size,
 			*off = umoff;
 			return 0;
 		}
-		D_ERROR("Reserve "DF_U64" from SCM failed.\n", size);
+
+		now = daos_gettime_coarse();
+		if (now - ioc->ic_cont->vc_io_nospc_ts > VOS_NOSPC_ERROR_INTVL) {
+			daos_size_t	scm_used = 0, scm_active = 0;
+
+			rc = pmemobj_ctl_get(vos_cont2pool(ioc->ic_cont)->vp_umm.umm_pool,
+					     "stats.heap.run_allocated", &scm_used);
+
+			rc = pmemobj_ctl_get(vos_cont2pool(ioc->ic_cont)->vp_umm.umm_pool,
+					     "stats.heap.run_active", &scm_active);
+
+			D_ERROR("Reserve "DF_U64" from SCM failed, run_allocated: "
```

Comment:
minor: should we ignore if above pmeobj_ctl_get return failed for some reasons.

Reply:
Yes, the pmemobj_ctl_get() failures are actually ignored, no matter if they succeeded or not, the error message will be printed. (if they failed, 0 run_allocated/active will be printed).